ml-operator/templates/image-watcher.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: image-watcher
+  labels:
+    app: image-watcher
+    app.kubernetes.io/name: image-watcher
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1 # exeed the replicas by 1 pod
+  selector:
+    matchLabels:
+      app: image-watcher
+  template:
+    metadata:
+      labels:
+        app: image-watcher
```

Comment:
Please add the `REC` labels as per Helm best practice guide: https://helm.sh/docs/chart_best_practices/labels/#standard-labels

I would actually recommend this approach:

1. Add these template functions to your `_helpers.tpl`:

```
{{/*
Create chart name and version as used by the chart label.
*/}}
{{- define "ml-operator.chart" -}}
{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
{{- end }}

{{/*
Common labels
*/}}
{{- define "ml-operator.labels" -}}
helm.sh/chart: {{ include "ml-operator.chart" . }}
{{ include "ml-operator.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- end }}

{{/*
Selector labels
*/}}
{{- define "ml-operator.selectorLabels" -}}
app.kubernetes.io/name: {{ include "ml-operator.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
```

2. Use the templates as such in the following service example:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: image-watcher
  labels:
    app: image-watcher
    {{- include "ml-operator.labels" . | nindent 4 }}
spec:
  replicas: 1
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 1 # exeed the replicas by 1 pod
  selector:
    matchLabels:
        {{- include "ml-operator.selectorLabels" . | nindent 6 }}
  template:
    metadata:
      labels:
        {{- include "ml-operator.labels" . | nindent 6 }}
```

I will be updating all charts to follow this approach once I start the "re-factoring dependencies".